src/ghga_transpiler/config.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Module to process config file"""
+
+from collections import Counter
+from collections.abc import Callable
+
+from pydantic import BaseModel, ConfigDict, Field, computed_field, model_validator
+
+from .exceptions import DuplicatedName
+from .transformations import to_attributes, to_list, to_snake_case, to_snake_case_list
+
+
+class ColumnProperties(BaseModel):
+    """A data model for column properties"""
+
+    model_config = ConfigDict(populate_by_name=True)
+
+    sheet_name: str = Field(..., alias="sheet")
+    column_name: str = Field(..., alias="column")
+    multivalued: bool
+    type: str
+    ref_class: str | None
+    ref_id: str | None = Field(..., alias="ref_class_id_property")
+    enum: bool
+    required: bool
+
+    @computed_field  # type: ignore [misc]
+    @property
+    def transformation(self) -> Callable | None:
+        """Assigns transformation function based on column properties"""
+        if self.multivalued and self.enum:
+            return to_snake_case_list()
```

Comment:
```suggestion
            return to_snake_case_list
```
You are returning the return values from calling these functions here, not the functions themselves. Same with the others.

Reply:
These functions return another Callable that does the transformation itself. i.e.` to_snake_case_list()` returns the function `snake_case_list`.

Reply:
I see, in that case just import `snake_case_list` and assign it.

Reply:
I think the current one is a cleaner design. We have a clear pattern in transformations.py. The function calling the other function that is doing the job does not have any positional arguments. Positional arguments are handled in the return value of those functions that are also a function. I think we should keep this pattern.